core/src/main/java/com/google/errorprone/bugpatterns/UnusedMethod.java

```diff
@@ -97,30 +98,68 @@ public final class UnusedMethod extends BugChecker implements CompilationUnitTre
   private static final ImmutableSet<String> EXEMPTING_METHOD_ANNOTATIONS =
       ImmutableSet.of(
           "com.fasterxml.jackson.annotation.JsonCreator",
+          "com.fasterxml.jackson.annotation.JsonValue",
           "com.google.inject.Provides",
           "com.google.inject.Inject",
           "com.google.inject.multibindings.ProvidesIntoMap",
           "com.google.inject.multibindings.ProvidesIntoSet",
           "com.tngtech.java.junit.dataprovider.DataProvider",
+          "jakarta.annotation.PreDestroy",
+          "jakarta.annotation.PostConstruct",
+          "jakarta.inject.Inject",
+          "jakarta.persistence.PostLoad",
+          "jakarta.persistence.PostPersist",
+          "jakarta.persistence.PostRemove",
+          "jakarta.persistence.PostUpdate",
+          "jakarta.persistence.PrePersist",
+          "jakarta.persistence.PreRemove",
+          "jakarta.persistence.PreUpdate",
           "javax.annotation.PreDestroy",
           "javax.annotation.PostConstruct",
           "javax.inject.Inject",
           "javax.persistence.PostLoad",
+          "javax.persistence.PostPersist",
+          "javax.persistence.PostRemove",
+          "javax.persistence.PostUpdate",
+          "javax.persistence.PrePersist",
+          "javax.persistence.PreRemove",
+          "javax.persistence.PreUpdate",
           "org.apache.beam.sdk.transforms.DoFn.ProcessElement",
           "org.aspectj.lang.annotation.Pointcut",
+          "org.aspectj.lang.annotation.After",
           "org.aspectj.lang.annotation.Before",
           "org.springframework.context.annotation.Bean",
           "org.testng.annotations.AfterClass",
           "org.testng.annotations.AfterMethod",
           "org.testng.annotations.BeforeClass",
           "org.testng.annotations.BeforeMethod",
           "org.testng.annotations.DataProvider",
+          "org.junit.After",
```

Comment:
Could you explain why these junit methods need to be exempted? I'd expect that they would always be public, so wouldn't be flagged.

Reply:
The reason for that is that `org.junit.AfterClass` and `org.junit.BeforeClass` were already in the list, so we wanted to make the list complete and consistent. 

_However_, based on the Javadoc and a local test, it _doesn't_ make sense to have them as they are indeed required to be `public`. 

Therefore we can omit the following annotations from the list:
- `org.junit.After`
- `org.junit.AfterClass`
- `org.junit.Before`
- `org.junit.BeforeClass`

Will push a commit to remove those from the list.

Reply:
Rebased and added a commit @graememorgan.